### PR TITLE
Pull request response

### PR DESCRIPTION
**Pull request by enochtsang**
@bseto @JordanHeinrichs @mpminardi

### CODE REVIEW

File: How-To-Pull-Request/README.adoc [hunk 1]
```diff
@@ -17,4 +19,13 @@ Here are the steps to contributing to a UCSolarCarTeam repository. The repositor
 .  Write a title and description of what the pull request is and includes.
 .  Wait for people to review the pull-request, respond accordingly and let someone else accept the merge. Even if you have the rights, DO NOT merge the pull-request yourself.
 
+== How to Respond to a Pull Request
+
+If you have been assigned to a pull request, you are responsible for making sure the code is tested and clean before merging. On top of those technical responsibilities, you are also responsible for responding to a pull request in a caring fashion. To promote pull request responses that are kind and helpful, we will have 4 different stages in responding to a pull request.
```
Comment (JordanHeinrichs) [inline How-To-Pull-Request/README.adoc]: It might be good to mention something about responding to a pull request in a timely fashion.

Comment (enochtsang) [inline How-To-Pull-Request/README.adoc]: What should I write? I don't really want to give a hard time to answer a pull request by.

Comment (JordanHeinrichs) [inline How-To-Pull-Request/README.adoc]: I think 3 or 4 days should be enough a week max to at least comment on it. I think it is a disservice to the person(s) making the request if they put the work in to get it by a deadline and no ones accepts it. If you don't want to put a hard deadline in the guide maybe put as soon as possible or something like that.

**Comment by JordanHeinrichs** [comment]
I would be for merging this in. I will wait till Byron and Mario look at it though.

File: How-To-Pull-Request/README.adoc [hunk 2]
```diff
@@ -17,4 +19,13 @@ Here are the steps to contributing to a UCSolarCarTeam repository. The repositor
 .  Write a title and description of what the pull request is and includes.
 .  Wait for people to review the pull-request, respond accordingly and let someone else accept the merge. Even if you have the rights, DO NOT merge the pull-request yourself.
 
+== How to Respond to a Pull Request
+
+If you have been assigned to a pull request, please respond as soon as possible, in general responding to a pull request should have priority over implementing other features. You are responsible for making sure the code is tested and clean before merging. On top of those technical responsibilities, you are also responsible for responding to a pull request in a caring fashion. To promote pull request responses that are kind and helpful, we will have 4 different stages in responding to a pull request.
```
Comment (bseto) [inline How-To-Pull-Request/README.adoc]: Just for commenting purposes, it is a good practice to put each sentence in its own line so I can comment on a specific sentence. When you put a sentence on a new line, asciidoc won't make it a new line unless you have the "+" so it will still appear like a normal paragraph.

Comment (enochtsang) [inline How-To-Pull-Request/README.adoc]: Do we want to change this for all the READMEs?

File: How-To-Pull-Request/README.adoc [hunk 3]
```diff
@@ -17,4 +19,13 @@ Here are the steps to contributing to a UCSolarCarTeam repository. The repositor
...
+
+. *Is the purpose of the pull request sound?* Do the overall changes make sense? 
+. *Is the code properly structured?* Will the code be maintainable in the future?
+. *Are there any bugs or glitches?* Test the code yourself to make sure it works.
+. *Is coding style good?* Does the code conform to how the rest of the project is coded? Does it follow the UCSolarCarDevelopmentGuide style guide? 
 
+Especially for newcomers, make sure the pull request passes one stage at a time. Once all the stages have been passed, then the pull request should be ready to be merged!
```
Comment (bseto) [inline How-To-Pull-Request/README.adoc]: This is good :)! However, since I read the original article, I already understand that the intent of this method is to review code in a couple "passes" or stages. I don't believe someone who is skimming over this would understand   the intent by reading this sentence.

I think the sentence should emphasize on not doing the code review in one go, but rather in a few passes.

File: How-To-Pull-Request/README.adoc (hunk 2 above)
Comment (bseto) [inline How-To-Pull-Request/README.adoc]: I think that would be good. I am 100% guilty of this too as I did not learn about this until recently :P . 
But ya, if i was trying to comment on the last sentence on your paragraph, I have no choice but to comment on the whole paragraph atm. (I don't have any comments for this paragraph other than the styling mentioned above btw so don't worry haha).

Comment (enochtsang) [inline How-To-Pull-Request/README.adoc]: We won't include it in this pull request, but can you make a different pull request for this change?

Comment (bseto) [inline How-To-Pull-Request/README.adoc]: Yep. I can do that later

**Comment by bseto** [comment]
Looks good! Have Mario look at it and then merge it in after.

File: How-To-Pull-Request/README.adoc [hunk 4]
```diff
@@ -17,4 +19,13 @@ Here are the steps to contributing to a UCSolarCarTeam repository. The repositor
...
 .  Wait for people to review the pull-request, respond accordingly and let someone else accept the merge. Even if you have the rights, DO NOT merge the pull-request yourself.
 
+== How to Respond to a Pull Request
+
+If you have been assigned to a pull request, please respond as soon as possible, in general responding to a pull request should have priority over implementing other features. You are responsible for making sure the code is tested and clean before merging. On top of those technical responsibilities, you are also responsible for responding to a pull request in a caring fashion. To promote pull request responses that are kind and helpful, we will have 4 different stages in responding to a pull request.
+
+. *Is the purpose of the pull request sound?* Do the overall changes make sense? 
+. *Is the code properly structured?* Will the code be maintainable in the future?
+. *Are there any bugs or glitches?* Test the code yourself to make sure it works.
+. *Is coding style good?* Does the code conform to how the rest of the project is coded? Does it follow the UCSolarCarDevelopmentGuide style guide?
```
Comment (mpminardi) [inline How-To-Pull-Request/README.adoc]: "Is coding style good?" sounds really odd. Maybe change to "Is the coding style correct?"

File: How-To-Pull-Request/README.adoc (hunk 2 above)
Comment (mpminardi) [inline How-To-Pull-Request/README.adoc]: Mega nit-picky, but replace the period at the end of this paragraph / on line 3 with colons.

**Comment by JordanHeinrichs** [comment]
Seeing that all comments were addressed I will merge it in.
